source/game.programme.newsevent.bmx

```diff
@@ -293,86 +323,94 @@ Type TNewsEventCollection
 
 	Method GetNewsHistory:TNewsEvent[](limit:Int=-1)
 		If limit = -1
-			Return newsEventsHistory[.. newsEventsHistoryIndex]
+			Return newsEventsHistory
 		Else
 			Return newsEventsHistory[Max(0,newsEventsHistoryIndex-limit) .. newsEventsHistoryIndex]
 		EndIf
 	End Method
 
 
 	'returns (and creates if needed) a list containing only follow up news
-	Method GetFollowingNewsList:TList(genre:Int=-1)
+	Method GetFollowingNewsList:TObjectList(genre:Int=-1)
 		'create if missing
 		If Not _followingNewsEvents Then _InvalidateFollowingNewsEvents()
 
-		If Not _followingNewsEvents[genre+1]
-			_followingNewsEvents[genre+1] = CreateList()
+		If Not _followingNewsEventsValid[genre+1]
```

Comment:
genre -1 ist doch "egal", sollten dann für -1 nicht alle Listen auf valid gesetzt werden? Oder werden die Events doppelt gehalten (in liste 0 und der genrespezifischen)?

Reply:
Die following-funktion wird noch gar nicht genutzt, gleiches gilt fuer genrespezifische upcoming-Getternutzung.

Aber ja, bei -1 sollten alle gesetzt werden...